### ISSUE TITLE
allow training fraction = 1 for TrainTestSplit pipeline

### ISSUE THREAD

**Issue by kzecchini** [ww-tech/primrose#38]
I think there may be a case where we want to train all of the data and use the same pipeline so there is no need to change code at all or run a different pipeline. This is a simple change, since there is already a check to see if test_data is empty. The logic can be incorporated in this function in `TrainTestSplit` class:

```
def _train_test_split(self, data):
        """Split data into test/train sets
        
        Returns:
            train_data_to_transform (DataFrame)
            
            test_data_to_transform (DataFrame)
        
        """
        logging.info("Splitting data into testing and training sets.")

        if 'target_variable' in self.node_config:
            data_train, data_test, target_train, target_test = train_test_split(
                data[self.features(data)],
                data[self.node_config['target_variable']],
                test_size=(1.0 - float(self.node_config['training_fraction'])),
                random_state=self.node_config['seed'])

            # re-merge training and target data into a single dataframe for transforming
            train_data_to_transform = pd.concat([data_train, target_train], axis=1)
            test_data_to_transform = pd.concat([data_test, target_test], axis=1)

        else:
            data_train, data_test = train_test_split(
                data[self.features(data)],
                test_size=(1.0 - float(self.node_config['training_fraction'])),
                random_state=self.node_config['seed'])

            train_data_to_transform = data_train
            test_data_to_transform = data_test

        logging.info('Training data rows: {}, Testing data rows: {}'.format(len(train_data_to_transform),
                                                                            len(test_data_to_transform)))

        return train_data_to_transform, test_data_to_transform
```

If `test_data_to_transform` is empty, then the `fit_transform` function already has logic built in to not run any transformation.

**Comment by briangrahamww** [comment]
Closed by #39

**Comment by briangrahamww** [comment]
Not sure why this was reopened
